compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/AbstractNessieApiHolder.java

```diff
@@ -129,7 +133,12 @@ protected static AutoCloseable createNessieClient(ClassLoader classLoader, Clien
           };
       builderInstance = fromConfigMethod.invoke(builderInstance, getCfg);
 
-      Class<?> targetClass = classLoader.loadClass(clientKey.getType().getName());
+      Class<?> targetClass;
+      try {
+        targetClass = classLoader.loadClass(clientKey.getType().getName());
+      } catch (ClassNotFoundException e) {
```

Comment:
Is this one to just have `null` for `NessieApiV2` w/ older clients? Should it be guarded using a check of `clientKey.getVersion()` against a constant in `Version`?

Reply:
Yes. Since injection is filed-based we have to provide a value for `NessieApiV2` fields in test classes even when the test env. is set up for V1.

I'll add a version check.

Reply:
I've refactored to a non-null proxy producing specific exceptions when/if the unsupported API is _accessed_ (in c3a1ee881d967c2ef244e1d11fd4a6de24ca2f33)... How does that look?